packages/hardhat/hardhat.config.js

```diff
@@ -141,6 +141,9 @@ module.exports = {
       },
     },
   },
+  // Set API key for etherscan/polygonscan/mumbai.polygonscan
+  // Endpoint will be chosed based on default network
+  // TODO: Replace this hardcode with a switch on defaultNetwork
   etherscan: {
     apiKey: '43JVMCI3EJCZSECDWX99KRZRYWDX3H7MCH'
```

Comment:
whose key is this? should we go ahead and migrate this to env?

Reply:
Not mine :p but it's a block explorer key, strictly read-only.
Yes on the env, there's infura keys in this file and a few other places. Can be a different PR collecting all these keys.

Reply:
I can support that